WLC/code_executor/abstract_executor.py

```diff
@@ -0,0 +1,89 @@
+import logging
+
+import docker
+import os
+from hackerrank.HackerRankAPI import HackerRankAPI
+from image_segmentation.preprocessor import Preprocessor
+
+from ..utils.azure import WLCAzure
+from ..ocr.picture_ocr import PictureOCR
+
+LOGGER = logging.getLogger()
+
+DEFAULT_DOCKER_PORT = "2375"
+
+
+class AbstractCodeExecutor:
+    NO_OUTPUT = "Program did not generate any output!"
+
+    def __init__(self, language, fixer, ip="", port=""):
+        self.language = language
+        self.fixer = fixer
+
+        if ip and port:
+            self.client = docker.DockerClient(base_url="tcp://{}:{}".format(ip, port))
+            self.force_local = False
+        else:
+            self.force_local = True
```

Comment:
Make it `True` by default and only change it in the if condition?

Reply:
Changed